src/data/field.rs

```diff
@@ -0,0 +1,116 @@
+//! Field parsing.
+
+use core::fmt::{self, Display, Formatter};
+
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, IdentFragment, ToTokens, TokenStreamExt};
+use syn::{Attribute, FieldsNamed, FieldsUnnamed, Ident, Index, Result};
+
+use crate::{FieldAttr, Skip, Trait};
+
+/// Struct, union, struct variant or tuple variant field.
+#[cfg_attr(test, derive(Debug))]
+pub struct Field<'a> {
+	/// Attributes.
+	pub attr: FieldAttr,
+	/// [`struct@Ident`] or [`Index`] for this field.
+	pub member: Member<'a>,
+	/// [`struct@Ident`] used as a Temporary variable for destructuring `self`.
+	pub self_ident: Ident,
+	/// [`struct@Ident`] used as a Temporary variable for destructuring `other`.
+	pub other_ident: Ident,
+}
+
+/// Borrowed version of [`syn::Member`], to avoid unnecessary allocations.
+#[cfg_attr(test, derive(Debug))]
+pub enum Member<'a> {
+	/// Named field.
+	Named(&'a Ident),
+	/// Unnamed field.
+	Unnamed(Index),
+}
+
+impl IdentFragment for Member<'_> {
+	fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+		Display::fmt(&self, f)
+	}
+}
+
+impl ToTokens for Member<'_> {
+	fn to_tokens(&self, tokens: &mut TokenStream) {
+		match self {
+			Member::Named(ident) => tokens.append_all(Some(ident)),
+			Member::Unnamed(index) => tokens.append_all(Some(index)),
+		}
+	}
+}
+
+impl Display for Member<'_> {
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		match self {
+			Member::Named(ident) => f.write_fmt(format_args!("{}", ident)),
```

Comment:
is there a reason for not using the write! macro?

Reply:
I actually have no clue, I thought using `format_args` is cheaper, but now I think that's just nonsense. I will look into it.

Reply:
Actually, what I wrote is exactly what `write!` does under the hood. Fixed!

Reply:
yeah I thought so thats why I was a bit confused and thought there is some msrv going on?